### PR TITLE
backport _poll

### PR DESCRIPTION
**Pull request by brian-armstrong-discord**

### CODE REVIEW

**Comment by goakley** [comment]
Were you able to try out this code on a box?

File: kombu/utils/eventio.py [hunk 1]
```diff
@@ -176,6 +186,68 @@ def close(self):
...
+        except socket.error as exc:
+            # we don't know the previous fd of this object
+            # but it will be removed by the next poll iteration.
+            if getattr(exc, 'errno', None) in SELECT_BAD_FD:
+                return fd
+            raise
+        self._quick_unregister(fd)
+        return fd
+
+    def poll(self, timeout, round=math.ceil,
```
Comment (goakley) [inline kombu/utils/eventio.py:222]: `import math`

**Comment by brian-armstrong-discord** [comment]
Nope, I did not

**Review by goakley — CHANGES_REQUESTED** [review]
Would recommend putting this in one of the worker's virtualenv and seeing if it starts pulling work successfully.

**Pull request merged**